gpflow/likelihoods/base.py

```diff
@@ -286,6 +288,62 @@ def _variational_expectations(self, Fmu, Fvar, Y):
         raise NotImplementedError
 
 
+class QuadratureLikelihood(Likelihood):
+    def __init__(self, num_gauss_hermite_points: int = 20, **kwargs):
+        super().__init__(**kwargs)
+        self.num_gauss_hermite_points = num_gauss_hermite_points
+        self._quadrature = None
+
+    @property
+    def quadrature(self):
+        if self._quadrature is None:
```

Comment:
Just a stylistic point, others may disagree, but I find this sort of precondition checking easier to read if you invert the if statements so that the condition and what happens if it fails are grouped together, e.g. 

`

if self._quadrature is None:
        return self._quadrature

if self.latent_dim is None:
    raise Exception(
        "latent_dim not specified. "
        "Either set likelihood.latent_dim directly or "
        "call a method which passes data to have it inferred."
    )

with tf.init_scope():
    self._quadrature = NDiagGHQuadrature(self.latent_dim, self.num_gauss_hermite_points)

`

That way you end up with fewer levels of nesting, and it's easier to separate the precondition checking from what the method _really_ does.